core/core-impl/src/main/java/ai/timefold/solver/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -1036,22 +1072,64 @@ public long getMaximumValueCount(Solution_ solution) {
                 .orElse(0L);
     }
 
+    public record ProblemScaleTracker(long logBase,
```

Comment:
This is abusing the record a bit, isn't it? Records are supposed to be immutable, and to get that, we introduce deeply mutable types. :-) Personally, I think this should just be a class, which in turn would allow us to get rid of the `MutableLong`.